hypothesis/lms
# Fetch canvas groups rosters using the names and roles API

For:

- https://github.com/hypothesis/lms/issues/6902


Canvas provides an extension to the names and roles API that allows to fetch the members of a group.

See:

https://canvas.instructure.com/doc/api/names_and_role.html#method.lti/ims/names_and_roles.group_index



### Testing


- Launch https://hypothesis.instructure.com/courses/319/assignments/3336
- Check who belongs to which group on 

https://hypothesis.instructure.com/courses/319/groups#tab-215

- In make shell, fetch the roster for the groups:



```
segments = db.query(models.LMSSegment).join(models.LMSCourse).where(models.LMSSegment.type=="canvas_group", models.LMSCourse.lti_context_id=='6ed5aff92b5257ab607ec9c629844ea8586fde68').all()

roster_service = request.find_service(services.RosterService)
for segment in segments: roster_service.fetch_canvas_group_roster(segment)
tm.commit()
```

- Check the information on the DB:

```
select name, display_name from lms_segment_roster join lms_segment on lms_segment.id = lms_segment_id join lms_user on lms_user.id = lms_user_id;
  name   |          display_name
---------+--------------------------------
 Group 1 | Hypothesis 101 Student Student
 Group 2 | AC Student 1 1
 Group 2 | AC Student 2 2
 Group 2 | AC Student 3 3
(4 rows)
```
